src/shipment/controller.js

```diff
@@ -194,15 +195,126 @@ async function createOne(req, res) {
-      const promises = manifest.map((x) =>
-        PackingSlip.updateOne({ _id: x }, { $set: { shipment: shipment._id } })
-      );
+      const promises = manifest.map( async (x) => {
+        const packingSlip = await PackingSlip.findOne({ _id: x });
+
+        orderNumbers.add(packingSlip.orderNumber)
+        if ( customerId === undefined ) customerId = packingSlip.customer.toString();
+
+        //build itemsShipped object
+        for ( const i of packingSlip.items ) {
+          itemsShipped.push( i.item.toString() );
+        }
+
+        packingSlip.shipment = shipment.id;
+        packingSlip.save();
+      } );
 
       customerDoc.numShipments = numShipments + 1;
       promises.push(customerDoc.save());
 
       await Promise.all(promises);
+      const orderNumbersArr = Array.from(orderNumbers);
+
+      //aggregation
+      const agg = [
+        { $match: {
+          customer: new ObjectId(customerId),
+          isPastVersion: false,
+        } },
+        { $unwind: '$manifest'},
+        { $lookup: {
+          from: 'packingSlips',
+          localField: 'manifest',
+          foreignField: '_id',
+          as: '_manifest'
+        } },
+        { $unwind: '$_manifest' },
+        { $match: {
+          $expr: {
+            $in: [ '$_manifest.orderNumber', orderNumbersArr]
+          }
+        } },
+        { $unwind: '$_manifest.items' },
+        { $match: {
+          $expr: { 
+            $in: [ { $toString: '$_manifest.items.item'} , itemsShipped ] 
+          }
+        } },
+        { $group: {
+          _id: '$_manifest.items.item',
+          orderNumber: { $first: '$_manifest.orderNumber' },
+          qtyShippedCustomer: { $sum: {
+            $cond: [
+              { $eq: ['$_manifest.destination', 'CUSTOMER'] },
+              '$_manifest.items.qty',
+              0
+            ]
+          } },
+          qtyShippedVendor: { $sum: {
+            $cond: [
+              { $eq: ['$_manifest.destination', 'VENDOR'] },
+              '$_manifest.items.qty',
+              0
+            ]
+          } }
+        } },
+        { $lookup: {
+          from: 'workorders',
+          let: { id: { $toString: '$_id' } },
+          pipeline: [
+            { $unwind: '$Items'},
+            { $match: {
+              $expr: {
+                $eq: [ '$$id', { $toString: '$Items._id' } ]
+              }
+            } },
+            { $project: {
+              'Items.Quantity': 1,
+              'Items.calcItemId': 1,
+            } },
+
+          ],
+          as: '_workOrder'
+        } },
+        { $unwind: '$_workOrder' },
+        { $addFields: {
+          totalQty: '$_workOrder.Items.Quantity',
+          calcItemId: '$_workOrder.Items.calcItemId'
+        } },
+        { $project: {
+          _workOrder: 0
+        } }
+      ];
+
+      const pipeline = await Shipment.aggregate(agg);
+
+      //loop through pipeline data and check if AT fields need set
+      const fields = {};
```

Comment:
move this inside the loop.

if there's multiple items in `pipeline[]` array your `fields{}` object is going to be stale since you're  not resetting it between loops.